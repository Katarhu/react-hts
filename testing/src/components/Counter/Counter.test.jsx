import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import renderer from "react-test-renderer";
import { createRenderer } from 'react-test-renderer/shallow';

import { Counter } from "./Counter";
import {CounterButton} from "../CounterButton/CounterButton";

const mockedIncrement = jest.fn();
const mockedDecrement = jest.fn();
const mockedSetCount = jest.fn();

jest.mock(
  "../../hooks/useCounter", () => ({
    useCounter: () => ({
      count: 0,
      increment: mockedIncrement,
      decrement: mockedDecrement,
      setCount: mockedSetCount,
    })
}));

describe('should match snapshot', () => {
  /*test('deep rendering', () => {
    const counter = renderer.create(<Counter/>);

    expect(counter).toMatchSnapshot();
  });*/

  test('shallow rendering', () => {
    const shallowRenderer = createRenderer();
    const counter = shallowRenderer.render(<Counter/>);

    expect(counter).toMatchSnapshot();
  });
});

describe('test elements', () => {
  beforeEach(() => {
    render(<Counter/>);
  });

  test("should call increment", () => {
    fireEvent.click(screen.getByText("+"));

    expect(mockedIncrement).toBeCalledTimes(1);
  });

  test("should call decrement", () => {
    fireEvent.click(screen.getByText("-"));

    expect(mockedDecrement).toBeCalledTimes(1);
  });

  test("input should have default value", () => {
    //const input = screen.getByRole("input");
    const input = screen.getByLabelText("input");

    expect(input).toHaveValue(0); //"0"
  });

  test("should call setCount", () => {
    const input = screen.getByLabelText("input");

    fireEvent.change(input, {
      target: {
        value: "15"
      }
    });

    expect(mockedSetCount).toBeCalledWith(15);
  });
});
