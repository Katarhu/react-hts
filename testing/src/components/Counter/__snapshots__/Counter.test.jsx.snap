// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot shallow rendering 1`] = `
<div>
  <input
    aria-label="input"
    onChange={[Function]}
    role="input"
    type="number"
    value={0}
  />
  <CounterButton
    onClick={[MockFunction]}
  >
    +
  </CounterButton>
  <CounterButton
    onClick={[MockFunction]}
  >
    -
  </CounterButton>
</div>
`;
